python/lance/data/convert/imagenet.py

```diff
@@ -0,0 +1,127 @@
+#  Copyright (c) 2022. Lance Developers
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+import itertools
+from pathlib import Path
+from typing import Dict, Generator, List, Optional
+
+import pyarrow as pa
+import pyarrow.dataset
+
+import lance
+from lance.types.image import Image, ImageArray
+
+__all__ = ["convert_imagenet_1k"]
+
+_SPLITS = ["train", "validation", "test"]
+
+
+def _to_record_batch(batch: Dict, label_names: List[str]) -> pa.RecordBatch:
+    """Convert a batch to RecordBatch."""
+    image_arr = ImageArray.from_images(batch["image"])
+    label_arr = pa.array(batch["label"], type=pa.int16())
+    names_arr = pa.DictionaryArray.from_arrays(
+        label_arr,
+        label_names,
+    )
+    split_arr = pa.DictionaryArray.from_arrays(
+        pa.array(batch["split"], type=pa.int8()), _SPLITS
+    )
+    id_arr = pa.array(batch["id"], pa.int32())
+    return pa.RecordBatch.from_arrays(
+        [id_arr, image_arr, label_arr, names_arr, split_arr],
+        ["id", "image", "label", "name", "split"],
+    )
+
+
+def _record_batch_gen(
+    batch_size: int = 1024, limit: Optional[int] = None
+) -> Generator[pa.RecordBatch, None, None]:
+    """Generator of RecordBatch."""
+    from datasets import load_dataset
+
+    example_id = 1
+    splits = ["train", "validation", "test"]
+    for split in splits:
+        hg_dataset = load_dataset("imagenet-1k", split=split, streaming=True)
+        hg_features = hg_dataset.features
+        batch = {"image": [], "label": [], "split": [], "id": []}
+        if limit:
+            hg_dataset = itertools.islice(hg_dataset, limit)
+        for example in hg_dataset:
```

Comment:
nit: this should be "sample" not "example"

Reply:
ok, but it is called example in some frameworks https://www.tensorflow.org/api_docs/python/tf/train/Example

Reply:
Tensorflow? what's that =P

Fair enough. Overall it's just that we'll see a lot of generic usage of example so you have to read carefully to know whether it's "In this Example" or "In this example".